src/api/app.py

```diff
@@ -31,24 +66,63 @@
     allow_headers=["*"],
 )
 
+if proxy_target:
+    logging.info(f"Proxy target set to: {proxy_target}")
+    @app.api_route("/{path:path}", methods=["GET", "POST", "PUT", "DELETE", "PATCH", "OPTIONS", "HEAD"])
+    async def proxy(request: Request, path: str):
+        # Build safe target URL
+        path_no_prefix = f"/{path.lstrip('/')}".removeprefix(API_ROUTE_PREFIX)
+        target_url = f"{proxy_target.rstrip('/')}/{path_no_prefix.lstrip('/')}".rstrip("/")
```

Comment:
maybe we should factor this out into a new function and use assert to verify an example?